tensorflow/java/src/main/java/org/tensorflow/Operation.java

```diff
@@ -97,6 +99,46 @@ public Output output(int idx) {
     return new Output(this, idx);
   }
 
+  @Override
+  public int hashCode() {
+    Graph.Reference r = graph.ref();
+    try {
+      // xor the first and last 32 bits of the underlying 64-bit pointer value.
+      return (int) (unsafeNativeHandle ^ (unsafeNativeHandle >>> 32));
```

Comment:
- A new object is created regardless, no? (`graph.ref()` is creating a new object), so the same for both cases?
- The exception: That seems fine. IIUC, the only contract is that if `a.equals(b)` then `a.hashCode() == b.hashCode()`, right? So, throwing an exception in `hashCode()` doesn't really help anyway?